daffodil-core/src/main/scala/org/apache/daffodil/core/runtime1/ElementBaseRuntime1Mixin.scala

```diff
@@ -89,17 +89,19 @@ trait ElementBaseRuntime1Mixin { self: ElementBase =>
     //
     // For complex elements with specified length, value length is captured in
     // the specified length parsers, since they handle skipping unused
-    // element regions. For complex elements, this means lengthKind is not
+    // element regions, and the length capturing has to be done before that.
+    // For complex elements, this means lengthKind is not
     // implicit or delimited.
     //
     // So for these cases we do not want to capture value length with the
     // Capture{Start,End}OfValueLengthParsers, since those lengths are captured
     // by the value parsers
-    val capturedByParsers =
-      (isSimpleType && (impliedRepresentation == Representation.Text || lengthKind == LengthKind.Delimited)) ||
+    val capturedByValueParsers =
+      (isSimpleType && (
+        primType == PrimType.String || lengthKind == LengthKind.Delimited)) ||
```

Comment:
I'm wondering if this change is correct?

For example, say we have this element:
```xml
<xs:element name="foo" type="xs:int" dfdl:representation="text" dfdl:trimKind="padChar" dfdl:lengthKind="explicit" dfdl:length="10" ... />
```
So a fixed length text integer with padding. In this case I think what Daffodil does is it create a String parser to parse the fixed length string and remove padding, and then creates another parser to convert that string to an actual integer.

So in that case, even though the primType is not String, I think the String parser will still be used to capture the value length after padding is removed. So I *think* `impliedRepresentation == Text` is still needed?